source/extensions/omni.isaac.lab/omni/isaac/lab/managers/reward_manager.py

```diff
@@ -13,28 +13,39 @@
 from typing import TYPE_CHECKING
 
 from .manager_base import ManagerBase, ManagerTermBase
-from .manager_term_cfg import RewardTermCfg
+from .manager_term_cfg import RewardGroupCfg, RewardTermCfg
 
 if TYPE_CHECKING:
     from omni.isaac.lab.envs import ManagerBasedRLEnv
 
 
+DEFAULT_GROUP_NAME = "reward"
+"""Default group name for backwards compatibility.
+
+If the user does not define any groups, the reward terms are added to this default group.
+"""
+
+
 class RewardManager(ManagerBase):
     """Manager for computing reward signals for a given world.
 
-    The reward manager computes the total reward as a sum of the weighted reward terms. The reward
-    terms are parsed from a nested config class containing the reward manger's settings and reward
-    terms configuration.
+    Rewards are organized into groups, where the total reward is computed independently for each group. Splitting into
+    groups is useful for multi-agent scenarios, where each agent has its own set of rewards or to compute an additional
+    oracle reward signal.
 
-    The reward terms are parsed from a config class containing the manager's settings and each term's
-    parameters. Each reward term should instantiate the :class:`RewardTermCfg` class.
+    Each reward group should inherit from the :class:`RewardGroupCfg` class. Within each group, each
+    reward term should instantiate the :class:`RewardTermCfg` class. The reward manager computes the total
+    reward for each group as a sum of the weighted reward terms.
 
     .. note::
 
         The reward manager multiplies the reward term's ``weight``  with the time-step interval ``dt``
         of the environment. This is done to ensure that the computed reward terms are balanced with
         respect to the chosen time-step interval in the environment.
 
+        For backwards compatibility, the reward manager also supports the old configuration format without
+        groups.
+
```

Comment:
Better to keep the note in a separate directive.

```suggestion

.. note::

        For backwards compatibility, the reward manager also supports the old configuration format without
        groups.

```